lib/samples/download_vector_tiles_to_local_cache/download_vector_tiles_to_local_cache_sample.dart

```diff
@@ -0,0 +1,386 @@
+  void startDownloadVectorTiles() async {
+    setState(() => _isJobStarted = true);
+    // Get the download area.
+    final downloadArea = downloadAreaEnvelope();
+    // Get the ArcGISVectorTiledLayer.
+    final vectorTileLayer = _mapViewController.arcGISMap!.basemap!.baseLayers[0]
+        as ArcGISVectorTiledLayer;
+    // Create an export vector tiles task.
+    final vectorTilesExportTask =
+        ExportVectorTilesTask.withUri(vectorTileLayer.uri!);
+    await vectorTilesExportTask.load();
+
+    // Get the cache directory to store the downloaded vector tiles
+    final resourceDirectory = await _getDownloadDirectory();
+    const vtpkName = 'myTileCacheDownload.vtpk';
+    final vtpkFile =
+        File('$resourceDirectory${Platform.pathSeparator}${vtpkName}');
+
+    // Create the default export vector tiles parameters,
+    // and shrink the area of interest by 10%.
+    final exportVectorTilesParameters =
+        await vectorTilesExportTask.createDefaultExportVectorTilesParameters(
+      areaOfInterest: downloadArea!,
+      maxScale: _mapViewController.scale * 0.1,
+    );
+
+    // Create the export vector tiles job.
+    _exportVectorTilesJob =
+        vectorTilesExportTask.exportVectorTilesWithItemResourceCache(
+      parameters: exportVectorTilesParameters,
+      vectorTileCacheUri: vtpkFile.uri,
+      itemResourceCacheUri: Uri.directory(resourceDirectory),
+    );
+
+    // Listen to the job's progress, status, and completion.
+    _exportVectorTilesJob?.onProgressChanged.listen((progress) {
+      setState(() => _progress = progress * 0.01);
+    });
+
+    // Listen to the job's status and handle the result.
+    _exportVectorTilesJob?.onStatusChanged.listen(
+      (status) {
+        if (status == JobStatus.failed) {
+          showDialog(
+            context: context,
+            builder: (context) => AlertDialog(
+              title:
+                  Text('Info', style: Theme.of(context).textTheme.titleMedium),
+              content: Text(
+                'Failed to download vector tiles:\n'
+                '${_exportVectorTilesJob?.error?.message}',
+              ),
+              actions: [
+                TextButton(
+                  onPressed: () {
+                    Navigator.of(context).pop();
+                  },
+                  child: const Text('Close'),
+                ),
+              ],
+            ),
+          );
+          setState(() => _isJobStarted = false);
+        } else if (status == JobStatus.succeeded) {
+          // If the job succeeded, load the downloaded vector tiles into the map view.
+          final result =
+              _exportVectorTilesJob?.result as ExportVectorTilesResult;
+          _loadExportedVectorTiles(result);
+          setState(() => _previewMap = true);
+          setState(() => _isJobStarted = false);
+        }
+      },
+    );
+
+    // Start the export vector tiles job.
+    _exportVectorTilesJob?.start();
```

Comment:
Normally for Jobs, we don't use `start()` and `onStatusChanged()`. Normally, we would structure it like this:
```dart
try {
  final result = await _exportVectorTilesJob?.run();
  // code to process result ...
} on ArcGISException catch (e) {
  showDialog(
    // code for an error dialog ...
 );
}
setState(() => _isJobStarted = false);
```

It's not wrong to use `start()` and `onStatusChanged()`, but I think it's clearer to use `run()` in a `try/catch` block.